src/figuresClasses.ts

```diff
@@ -1,11 +1,80 @@
-export interface Figure {}
+export type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+export interface Figure {
+  color: Color;
+  shape: Shape;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a >= c + b || b >= a + c || c >= b + a) {
```

Comment:
You can also add an additional check here.

```suggestion
  ) {
    checkSides('Triangle sides must be greater than 0', a, b, c);
  
    if (a >= c + b || b >= a + c || c >= b + a) {
```